documentation/src/main/asciidoc/topics/proc_setting_up_credential_keystores.adoc

```diff
@@ -56,3 +56,52 @@ Attributes in the `credential-reference` configuration are optional.
 * `store` is required only if you have multiple keystores.
 * `alias` is required only if the keystore contains multiple aliases.
 ====
+
+== Externalizing and masking credential keystore passwords
+
+Instead of providing your credential store’s master password in the clear, you can choose to provide that password using a pseudo credential store. You have the following options:
+
+=== External command
+
+External command using `java.lang.Runtime#exec(java.lang.String)`. If parameters are needed, they are supplied using a space-separated list of strings. An external command refers to any executable from the operating system, for example a shell script or an executable binary. The password is read from the standard output of the executed command.
+
+[discrete]
+=== External credentials
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/server_command_credentials.xml[]
+----
+
+.JSON
+[source,json,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::json/server_command_credentials.json[]
+----
+
+.YAML
+[source,yaml,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::yaml/server_command_credentials.yaml[]
+----
+
+=== MASK
```

Comment:
it would be nice to have the xml, yaml and json configuration on how to configure the credentials mask.

Reply:
Fixed